lib/Parser/Legacy/LimitedNumeric.pm

```diff
@@ -36,6 +36,12 @@ sub _check {
 
 sub class {'MINUS'};
 
+# Override the parent method because the class for this package is 'MINUS'
+# instead of 'UOP' as it is for the Parser::UOP::minus parent package.
+sub isNeg {
+  my $self = shift;
+  return $self->class eq 'MINUS' && $self->{uop} eq 'u-' && !$self->{op}{isInfinite};
```

Comment:
You can probably just `return !$self->{op}{isInfinite}`, here, since this object's class is always `MINUS` (unless someone makes a subclass of it that changes that, in which case, they will have to override this method anyway), and from `u-` (unless the context is remapped).

Reply:
Yeah, that is true.  Good point.